ush/python/pygfs/task/marine_analysis.py

```diff
@@ -42,452 +48,151 @@ def __init__(self, config):
-
-        # copy obs from COM_OBS to DATA/obs
-        for obs_file in obs_files:
-            logger.info(f"******* {obs_file}")
-            obs_src = os.path.join(self.task_config.COM_OBS, obs_file)
-            obs_dst = os.path.join(self.task_config.DATA, 'obs', obs_file)
-            logger.info(f"******* {obs_src}")
-            if os.path.exists(obs_src):
-                logger.info(f"******* fetching {obs_file}")
-                obs_files_to_copy.append([obs_src, obs_dst])
-            else:
-                logger.info(f"******* {obs_file} is not in the database")
+        - preparing the namelists for deterministic MOM6 and analysis geometry
+        - staging observations
+        - staging input YAMLs for SOCA utilities
+        - staging the deterministic backgrounds (middle of window)
+        - staging files and link directories from B-matrix job needed for deterministic analysis
+        - generating list of model pseudo states
+        - initializing all the JEDI applications required for the marine analysis
 
-        FileHandler({'copy': obs_files_to_copy}).sync()
+        Parameters
+        ----------
+        None
 
-    @logit(logger)
-    def _prep_scratch_dir(self: Task) -> None:
-        """Create and stage all the resources needed to run SOCA/JEDI, including the necesssary
-           directory structure to run the SOCA variational application
+        Returns
+        ----------
+        None
         """
-        logger.info(f"---------------- Setup runtime environement")
-
-        anl_dir = self.task_config.DATA
-
-        # create analysis directories
-        diags = os.path.join(anl_dir, 'diags')            # output dir for soca DA obs space
-        obs_in = os.path.join(anl_dir, 'obs')             # input      "           "
-        anl_out = os.path.join(anl_dir, 'Data')           # output dir for soca DA
-        FileHandler({'mkdir': [diags, obs_in, anl_out]}).sync()
 
         # stage fix files
         logger.info(f"Staging SOCA fix files from {self.task_config.SOCA_INPUT_FIX_DIR}")
         soca_fix_list = parse_j2yaml(self.task_config.SOCA_FIX_YAML_TMPL, self.task_config)
         FileHandler(soca_fix_list).sync()
 
         # prepare the deterministic MOM6 input.nml
+        logger.info(f"Preparing deterministic MOM6 input namelist")
         mdau.prep_input_nml(self.task_config)
 
         # prepare the input.nml for the analysis geometry
+        logger.info(f"Preparing analysis geometry input namelist")
         mdau.prep_input_nml(self.task_config, output_nml="./anl_geom/mom_input.nml",
                             simple_geom=True, mom_input="./anl_geom/MOM_input")
 
+        # fetch observations from COMROOT
+        # TODO(G.V. or A.E.): Keep a copy of the obs in the scratch fs after the obs prep job
+        logger.info(f"Staging observations from {self.task_config.COM_OBS}")
+        obs_list = self.jedi_dict['var'].render_jcb(self.task_config, 'soca_obs_staging')
```

Comment:
Curious what `soca_obs_staging` is referencing - it doesn't show up in any repos

Reply:
Ah, never mind, saw it in the` jcb-gdas` PR